tests/riotboot/tests/01-run.py

```diff
@@ -65,16 +72,26 @@ def testfunc(child):
     child.expect_exact('>')
     child.sendline("curslothdr")
     child.expect(r"Image Version: (?P<app_ver>0x[0-9a-fA-F]{8})\r\n")
-    current_app_ver = int(child.match.group("app_ver"), 16)
-
-    # Flash to both slots and verify basic functions
-    for version in [current_app_ver + 1, current_app_ver + 2]:
-        slotnum = slotnum ^ 1
-        flash_slot(slotnum, version)
-        assert_check_slot(child, slotnum, version)
+    app_ver = int(child.match.group("app_ver"), 16)
 
-    print("TEST PASSED")
+    # Set current slot number and app version
+    global current_app_ver
+    global current_slot_num
+    current_slot_num = slotnum
+    current_app_ver = app_ver
 
 
 if __name__ == "__main__":
-    sys.exit(run(testfunc))
+    res = 1
+    global current_app_ver
+    global current_slot_num
+    # set current slot number and app version
+    res = run(testcurrent)
+    # flash to both slots and verify
+    for version in [current_app_ver + 1, current_app_ver + 2]:
+        current_slot_num = current_slot_num ^ 1
+        flash_slot(current_slot_num, version)
+        current_app_ver = version
+        run(testfunc)
```

Comment:
The return res is not taken into account: so if testfunc fail, the call to this script won't fail since it exists using the return value of `run(testcurrent)`.